File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int get_index(int letter, int word, int sentence);
void get_grade(int index);
//index = 0.0588 * L - 0.296 * S - 15.8
// L / words * 100 = letters per 100 words
// S / words * 100 = sentences per 100 words
int main(void)
{
    string text = get_string("Text: ");
    //printf("%s\n", text);

    int L = count_letters(text);
    //printf("%i letters\n", L);

    int W = count_words(text);
    //printf("%i words\n", W);

    int S = count_sentences(text);
    //printf("%i sentences\n", S);

    int index = get_index(L, W, S);
    //printf("%i index\n", index);

    get_grade(index);
}

int count_letters(string text)//calculates number of letters in the text
{
    int letterCount = 0;
    int length = strlen(text);

    for (int i = 0; i < length; i++)
    {
        if (isalpha(text[i]))
            letterCount++;
    }

    return letterCount;
}

int count_words(string text)//calculates number of words in the text
{
    int wordCount = 1;
    int length = strlen(text);

    for(int i = 0; i <= length; i++)
    {
        if((int) text[i] == 32)
            wordCount++;
    }

    return wordCount;
}

int count_sentences(string text)//calculates number of sentences in the text
{
    int sentenceCount = 0;
    int length = strlen(text);

    for(int i = 0; i <= length; i++)
    {
        if((int) text[i] == 33 || (int) text[i] == 46 || (int) text[i] == 63)
            sentenceCount++;
    }

    return sentenceCount;
}

int get_index(int letter, int word, int sentence)
{
    float lpw = letter / (float) word * 100;
    //where LPW is letters per 100 words
    float spw = sentence / (float) word * 100;
    //where SPW is letters per 100 words

    return round(0.0588 * lpw - 0.296 * spw - 15.8); //Coleman-Liau index caluculator.
}

void get_grade(int index)
{
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }else{
        printf("Grade %i\n", index);
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string
import re


def main():
    # get text from user and store the length of the text
    text = get_string("Text: ")
    length = len(text)
    # printf("%s\n", text)

    # get number of letters in the text
    L = count_letters(text, length)
    # print(f"letters {L}")

    # get number of words in the text
    W = count_words(text)
    # print(f"word {W}")

    # get number of sentences in the text
    S = count_sentences(text, length)
    # print(f"sentences {S}")

    # get index from the formula
    index = get_index(L, W, S)
    # print(f"index {index}")

    # print the grade of the text
    get_grade(index)
    

def count_letters(text, length):
    letterCount = 0

    for i in range(length):
        if text[i].isalpha():
            letterCount += 1

    return letterCount


def count_words(text):
    wordCount = text.count(" ") + 1
    return wordCount


def count_sentences(text, length):
    count = 0

    for i in range(length):
        if re.match('[.?!]', text[i]):
            count += 1

    return count


def get_index(letter, word, sentence):

    # calculate letters per word
    lpw = letter / word * 100

    # calculate sentences per word
    spw = sentence / word * 100

    total = round(0.0588 * lpw - 0.296 * spw - 15.8)

    return total


def get_grade(index):
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


main()